web3/contract.py

```diff
@@ -589,38 +592,27 @@ def __getattr__(self, function_name):
     _return_data_normalizers = tuple()
 
     @classmethod
-    def _find_matching_fn_abi(cls, fn_name=None, args=None, kwargs=None):
-        filters = []
-
-        if fn_name:
-            filters.append(functools.partial(filter_by_name, fn_name))
-
-        if args is not None or kwargs is not None:
-            if args is None:
-                args = tuple()
-            if kwargs is None:
-                kwargs = {}
-
-            num_arguments = len(args) + len(kwargs)
-            filters.extend([
-                functools.partial(filter_by_argument_count, num_arguments),
-                functools.partial(filter_by_encodability, args, kwargs),
-            ])
-
-        function_candidates = filter_by_type('function', cls.abi)
-
-        for filter_fn in filters:
-            function_candidates = filter_fn(function_candidates)
+    def _prepare_transaction(cls,
+                             fn_name,
+                             fn_args=None,
+                             fn_kwargs=None,
+                             transaction=None):
 
-            if len(function_candidates) == 1:
-                return function_candidates[0]
-            elif not function_candidates:
-                break
+        return prepare_transaction(cls.abi,
+                                   cls.address,
+                                   cls.web3,
+                                   fn_name=fn_name,
+                                   fn_args=fn_args,
+                                   fn_kwargs=fn_kwargs,
+                                   transaction=transaction)
+                                   
```

Comment:
This line is causing the `flake8` errors. Deleting it will fix both errors. Once travis-ci passes, and we should be good to merge.